### PR TITLE
Ignore m.replace relations on state events, they're invalid

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/21851

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Ignore m.replace relations on state events, they're invalid ([\#2306](https://github.com/matrix-org/matrix-js-sdk/pull/2306)). Fixes vector-im/element-web#21851.

### CODE REVIEW

File: src/models/event.ts [hunk 1]
```diff
@@ -1336,6 +1336,10 @@ export class MatrixEvent extends TypedEventEmitter<EmittedEvents, MatrixEventHan
         if (this.isRedacted() && newEvent) {
             return;
         }
+        // don't allow state events to be replaced using this mechanism as per MSC2676
+        if (this.isState()) {
```
Comment (gsouquet) [inline src/models/event.ts:1340]: That looks pretty sensible to me. Any chance to validate that behaviour in a test?

**Comment by andybalaam** [comment]
I commented here: https://github.com/matrix-org/matrix-spec-proposals/pull/2676#pullrequestreview-946294047 since it seems a better place to discuss this than on this PR.

**Review by gsouquet — APPROVED** [review]
Code looks good to me 👍

**Comment by kerryarchibald** [comment]
Can be merged after https://github.com/matrix-org/matrix-react-sdk/pull/8390

**Pull request merged**